gtda/homology/rescaling.py

```diff
@@ -0,0 +1,333 @@
+    `X`. If each entry in `X` along axis 0 represents a distance matrix
+    :math:`D`, then the corresponding entry in the transformed array is the
+    distance matrix :math:`D'_{i,i+1} = factor\times D_{i,i+1}`. If the entries
+    in `X` represent point clouds, their distance matrices are first computed,
+    and then rescaled according to the same formula.
+
+    Parameters
+    ----------
+    metric : string or callable, optional, default: ``'euclidean'``
+        If set to ``'precomputed'``, each entry in `X` along axis 0 is
+        interpreted to be a distance matrix. Otherwise, entries are
+        interpreted as feature arrays, and `metric` determines a rule with
+        which to calculate distances between pairs of instances (i.e. rows)
+        in these arrays.
+        If `metric` is a string, it must be one of the options allowed by
+        :func:`scipy.spatial.distance.pdist` for its metric parameter, or a
+        metric listed in :obj:`sklearn.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`,
+        including "euclidean", "manhattan" or "cosine".
+        If `metric` is a callable function, it is called on each pair of
+        instances and the resulting value recorded. The callable should take
+        two arrays from the entry in `X` as input, and return a value
+        indicating the distance between them.
+
+    metric_params : dict or None, optional, default: ``None``
+        Additional keyword arguments for the metric function.
+
+    factor : float, optional, default: ``0.``
+        Factor by which to multiply the distance between consecutive
+        points.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means
+        using all processors.
+
+    Attributes
+    ----------
+    effective_metric_params_ : dict
+        Dictionary containing all information present in `metric_params`.
+        If `metric_params` is ``None``, it is set to the empty dictionary.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from gtda.homology import ConsecutiveRescaling
+    >>> X = np.array([[[0, 0], [1, 2], [5, 6]]])
+    >>> cr = ConsecutiveRescaling()
+    >>> X_rescaled = cr.fit_transform(X)
+    >>> print(X_rescaled.shape)
+    (1, 3, 3)
+
+    See also
+    --------
+    VietorisRipsPersistence
+
+    """
+    _hyperparameters = {'factor': [numbers.Number, (0., np.inf)]}
+
+    def __init__(self, metric='euclidean', metric_params=None, factor=0.,
+                 n_jobs=None):
+        self.metric = metric
+        self.metric_params = metric_params
+        self.factor = factor
+        self.n_jobs = n_jobs
+
+    def _consecutive_homology_distance(self, X):
+        Xm = pairwise_distances(X, metric=self.metric, n_jobs=1,
+                                **self.effective_metric_params_)
+
+        Xm.ravel()[
```

Comment:
What's wrong with
```
Xm[:, range(X.shape[1]-1), range(1, X.shape[1])] *= self.factor
```
?

Reply:
`IndexError: too many indices for array`

I could not make this kind of logic work well. The slicing is a bit weird, that's why I add to use `ravel`.

Reply:
`IndexError: too many indices for array`

I could not make this kind of logic work well. The slicing is a bit weird, that's why I add to use `ravel`.

Reply:
That's because I was being stupid thinking `Xm` was 3-dimensional!  But `X` here is 2-dimensional (a single sample), so the correct code should be
```
Xm[range(Xm.shape[0]-1), range(1, Xm.shape[0])] *= self.factor
```
See https://gist.github.com/ulupo/76e3a60284f174cd145b958156d5a7ab to see the effect.